source/MRCommonPlugins/Voxels/MRBinaryOperationsPlugin.h

```diff
@@ -36,10 +36,18 @@ class BinaryOperations : public StatePlugin, public SceneStateExactCheck<2, Obje
         Replace,
         Count
     };
-    void doOperation_(Operation op);
+    void doOperation_( Operation op, bool inPreview );
+
+    void onTransformChange();
 
     std::shared_ptr<ObjectVoxels> obj1_;
     std::shared_ptr<ObjectVoxels> obj2_;
+    boost::signals2::scoped_connection conn1_;
+    boost::signals2::scoped_connection conn2_;
+
+    std::shared_ptr<ObjectVoxels> previewRes_;
+    bool previewMode_ = false;
+    Operation previewOp_ = Operation::Union;
```

Comment:
should it be called simply `operation_`?

Reply:
![image](https://github.com/user-attachments/assets/5c2d55db-df13-4946-9013-5b36d36eeaf0)